src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1654,6 +1644,10 @@ void EmitBackreference(RegexNode node)
 
                 Label condition = DefineLabel();
                 Label body = DefineLabel();
+                Label charactersMatched = DefineLabel();
+                LocalBuilder backreferenceCharacter = _ilg!.DeclareLocal(typeof(char));
+                LocalBuilder currentCharacter = _ilg!.DeclareLocal(typeof(char));
+                LocalBuilder caseEquivalences = DeclareReadOnlySpanChar();
```

Comment:
We should only DeclareXx these if they'll actually be needed because this is IgnoreCase.